tools/compare_codegen.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+# SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+
+set -e
+set -o pipefail
+
+if (( $# > 2 ))
+then
+    echo "Usage: $0 [ compare_ref=origin/main ] [ out_dir=codegen_comparison ]"
+    exit 1
+fi
+
+comparetoref=${1:-origin/main}
+comparecommit=$(git describe --always --long $comparetoref)
+outdir=${2:-codegen_comparison}
+
+if [[ $(git status --porcelain --untracked-files=no) ]]
+then
+    echo "Must checkout main in order to compare. Commit changes before running this script."
+    exit 1
+fi
+
+# save current commit and current head so we can switch back to branch
+currentcommit=$(git describe --always --long)
+origcommit=$currentcommit
+orighead=$(git symbolic-ref --short HEAD)
+
+movecudafiles() {
+    find . -maxdepth 1 -name '__tmp_kernel*.cu' -exec mv '{}' $1 \;
+}
+
+cleanup() {
+    numkernels=$(find . -maxdepth 1 -name '__tmp_kernel*.cu' | wc -l)
+
+    if (( $numkernels > 0 ))
+    then
+        backupdir=$outdir/${currentcommit}-interrupted
+        echo "Interrupted. Backing up $numkernels .cu files to $backupdir"
+        mkdir -p $backupdir
+        movecudafiles $backupdir
+    fi
+
+    git switch $orighead
+}
+
+trap "cleanup" INT TERM EXIT
+
+collect_kernels() {
+    outdir=$1
+    commit=$2
+
+    git checkout $commit
+    currentcommit=$commit
+
+    testdir=$outdir/$commit/test
+    benchdir=$outdir/$commit/bench
+
+    # build in release mode so that everything runs a bit faster
+    if [[ -d $testdir/cuda && -d $benchdir/cuda ]]
+    then
+        # Test for output directories and return early if they exist. This
+        # avoids rebuilds when we are changing code and comparing repeatedly to
+        # the same earlier commit.
+        return
+    fi
+    python setup.py develop --cmake-only
+    ninja -j20 -C build nvfuser_tests nvfuser_bench
```

Comment:
Is there any reason for not just
```
python setup.py develop
```

Reply:
Not really, and actually I'm going to switch to that since we might also want to include python tests at some point.